mongo_database.py

```diff
@@ -0,0 +1,186 @@
+    status = StringField(required=True)
+    phase = StringField(required=True)
+    polling = BooleanField(required=True)
+    roomMaster = StringField(required=True)
+    votes = ListField(EmbeddedDocumentField(Vote))
+    gameMessages = ListField(EmbeddedDocumentField(GameMessage))
+    observerMessages = ListField(EmbeddedDocumentField(ObserverMessage))
+
+    def active_mafia(self, userId):
+        for player in self.players:
+            if player.userId == userId and player.role == 'mafia':
+                return player
+
+    def get_player(self, userId):
+        for player in self.players:
+            if player.userId == userId:
+                return player
+
+    def check_vote(self, userId):
+        for vote in self.votes:
+            if vote.userId == userId:
+                return True
+
+    @property
+    def hang(self):
+        max_count = 0
+        suspectIdList = []
+        is_execution = False
+        for i in self.players:
+            count = sum(vote.targetId == i.userId for vote in self.votes)
+            if count > max_count:
+                max_count = count
+                suspectIdList.clear()
+                suspectIdList.append(i.userId)
+            elif count == max_count:
+                suspectIdList.append(i.userId)
+        max_count = ceil(max_count)
+
+        if max_count >= ceil(self.players_left/2):
+            for player in self.players:
+                if player.userId == suspectIdList[0]:
+                    player.status = 'dead'
+                    self.votes.clear()
+                    self.gameMessages.append(
+                        GameMessage(primary='Execution', secondary=player.name + ' was hanged. Votes: ' + str(max_count) + ' Players left: ' + str(self.players_left)))
+                    self.observerMessages.append(
+                        ObserverMessage(primary='Execution', secondary=player.name + ' was hanged. Votes: ' + str(max_count) + ' Players left: ' + str(self.players_left)))
+                    self.save()
+                    return True
+        self.votes.clear()
+        self.gameMessages.append(
+            GameMessage(primary='Execution', secondary='Not enough votes, no one was hanged.'))
+        self.observerMessages.append(
+            ObserverMessage(primary='Execution', secondary='Not enough votes, no one was hanged.'))
+        self.save()
+        return False
+
+    @property
+    def doctor(self):
+        for player in self.players:
+            if player.role == 'doctor':
+                return player
+
+    @property
+    def detective(self):
+        for player in self.players:
+            if player.role == 'detective':
+                return player
+
+    @property
+    def players_left(self):
+        return sum(players.status == 'alive' for players in self.players)
+
+    @property
+    def mafia_left(self):
+        return sum(players.status == 'alive' and players.role == 'mafia' for players in self.players)
```

Comment:
this is minor, but singular is more semantic and i think more pythonic:
```suggestion
        return sum(player.status == 'alive' for player in self.players)

    @property
    def mafia_left(self):
        return sum(player.status == 'alive' and player.role == 'mafia' for player in self.players)
```

Reply:
oh yeah that was copy and paste haha